jina/drivers/rank.py

```diff
@@ -46,11 +46,10 @@ def _apply_all(self, docs: 'jina_pb2.Document', context_doc: 'jina_pb2.Document'
         match_idx = []
         query_chunk_meta = {}
         match_chunk_meta = {}
-        for c in docs:
-            for match in c.matches:
-                match_idx.append((match.parent_id, match.id, c.id, match.score.value))
-                query_chunk_meta[c.id] = pb_obj2dict(c, self.exec.required_keys)
-                match_chunk_meta[match.id] = pb_obj2dict(match, self.exec.required_keys)
+        for match in docs:
```

Comment:
I had to make this change @nan-wang @hanxiao, 

in the case of the added example:

When we are traversing on `matches`, all these `docs` will be matches from some `previous` indexer. Therefore they are the matches that we need to sort.

I am sure I am breaking some case here, but it seems to me that this has to be reviewed to cover all cases, or we are maybe needing a different Driver and not every case can be handled by this one.

Reply:
Done!